### PR TITLE
getRegionHtmlSelect does not have configuration

### PR DESCRIPTION
**Pull request by kandarp26**
### Description

When i use the getRegionHtmlSelect function in custom module there is not way to pass argument for create region drop down.  This arguments available in getCountryHtmlSelect in same way need to implement in getRegionHtmlSelect .

### Fixed Issues 
magento/magento2#7491:Method does not have configuration

### Manual testing scenarios
1. Created the address from customer account. 
2. Created the Order.
3. Created and Edit the customer from admin with address details. 

### Contribution checklist
   [X] Pull request has a meaningful description of its purpose
   [X] All commits are accompanied by meaningful commit messages
   [ ] All new or changed code is covered with unit/integration tests 
  [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=16445) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @kandarp26. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by kandarp26** [comment]
@magento-engcom-team give me 2.2 instance

**Comment by kandarp26** [comment]
@magento-engcom-team give me test instance

**Comment by magento-engcom-team** [comment]
Hi @kandarp26. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @kandarp26, here is your [new Magento instance](https://pr-16445.engcom.dev.magento.com).
Admin access: [https://pr-16445.engcom.dev.magento.com/admin](https://pr-16445.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Please make sure you are PR **author** or **assignee** to access the instance.

**Comment by apomili** [comment]
Hi Guys, I'm trying to add a css class to the selects created by the getCountryHtmlSelect  and getRegionHtmlSelect function.

If I'm not wrong there is no way to accomplish this  without extending data.php

**Comment by kandarp26** [comment]
Yes, We can improve this function. for css class also. Let's this pull request get process then we will update for css also .

File: app/code/Magento/Directory/Block/Data.php
Comment (avoelkl) [inline app/code/Magento/Directory/Block/Data.php]: Why not call this variable $id if it sets the id? $state is kind of misleading i think. In getCountryHtmlSelect $id = 'country'.

**Review by avoelkl — CHANGES_REQUESTED** [review]

File: app/code/Magento/Directory/Block/Data.php
Comment (kandarp26) [reply app/code/Magento/Directory/Block/Data.php]: I updated this changes but in code review it would be fail due to sort name for variable.

**Comment by kandarp26** [comment]
@avoelkl please let me know if any other changes required.

**Comment by kandarp26** [comment]
@avoelkl waiting for your response.

**Comment by kandarp26** [comment]
@ihor-sviziev This pull request in review from long time. please advice me what to do .

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-2804](https://jira.corp.magento.com/browse/ENGCOM-2804) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@kandarp26 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by ishakhsuvarov** [comment]
Hi @kandarp26 
To maximize backwards compatibility I would suggest to introduce another method, which would be capable of accepting configurable values. Original method would then just proxy to the new method with the default values. Do you think it's possible to implement that way?

**Comment by kandarp26** [comment]
@ishakhsuvarov : This function also work without any argument as all argument have default value. I don't think so other function required for it.

File: app/code/Magento/Directory/Block/Data.php
Comment (sidolov) [inline app/code/Magento/Directory/Block/Data.php]: We are not allowed to add new parameters to public methods according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/). Please, consider creating the new method and deprecate the old method.

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
Hi @kandarp26,
Right now static tests are failing, will you be able to fix the code?
Also you just marked `getRegionHtmlSelect()` as deprecated, but it's better to run new method inside it in order not to duplicate the same logic in these two methods

**Comment by sidolov** [comment]
Hi @kandarp26 , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by nikunjskd20** [comment]
Hi @sidolov, I have found this issue in the 2.3-develop branch. is this PR accepted? I have resolved this issue in my fork.

**Comment by ihor-sviziev** [comment]
Hi @nikunjskd20,
As you can see from last comment - pr was closed due to inactivity and wasn’t merged. Feel free to create PR

**Pull request closed without merging**
